app/src/lib/git/status.ts

```diff
@@ -35,7 +35,33 @@ export interface IStatusResult {
   readonly workingDirectory: WorkingDirectoryStatus
 }
 
-function convertToAppStatus(status: FileEntry): AppFileStatus {
+async function entryHasConflictMarkers(
+  repositoryPath: string,
+  entry: IStatusEntry,
+  status: FileEntry
+): Promise<boolean | null> {
+  // only conflicted files can have conflict markers
+  if (status.kind !== 'conflicted') {
+    return null
+  }
+
+  const args = ['diff', '--check', entry.path]
+  const { exitCode } = await spawnAndComplete(
+    args,
+    repositoryPath,
+    'diffCheck',
+    new Set([0, 2]),
+    // we don't care about the output
+    0
+  )
+  // 2 means conflict markers were found
+  return exitCode === 2
+}
+
+function convertToAppStatus(
+  status: FileEntry,
+  hasConflictMarkers: boolean | null
```

Comment:
does it make more sense to use a single type that bundles a `FileEntry` and a `boolean` here? it seemed like overkill to me but i'm not totally sure

Reply:
When using a `FileEntry` do you usually pass it with a `Bool`? If so, then it should probably be a type. If you only are using this `FileEntry` in fewer than 3 places, it's probably overkill.

Reply:
i think this is good then :) thanks!